RCB-539

guruxu opened the pull request:
- merge from 1.8 branch
- lombok on models
- simplified mixins
- clean-up deprecated etc.

Commit:
Preparing release. Version set to 1.8.0-SNAPSHOT

Commit:
[maven-release-plugin] prepare release rosette-api-java-binding-1.8.0

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
[maven-release-plugin] rollback the release of rosette-api-java-binding-1.8.0

Commit:
[maven-release-plugin] prepare release rosette-api-java-binding-1.8.0

Commit:
[maven-release-plugin] prepare for next development iteration

Katsuya-Tomioka committed:
Tej 1058 linking confidence rosapi 1.8 (#110)

* TEJ-1058: added linkingConfidence to EntityMention

* TEJ-1058: test data

Katsuya-Tomioka committed:
Tej 1058 sentiment (#112)

* [maven-release-plugin] prepare release rosette-api-java-binding-1.8.1

* [maven-release-plugin] prepare for next development iteration

* TEJ-1058: added linking confidence in EntitySentiment.

Katsuya-Tomioka committed:
[maven-release-plugin] prepare release rosette-api-java-binding-1.8.2

Katsuya-Tomioka committed:
[maven-release-plugin] prepare for next development iteration

Commit:
RCB-539: first drop of everything appears working
- Merge branch 'rosapi-1.8' with master
- lombok'ed all models
- jackson mixin cleaned up, mostly
- many other touch ups

Commit:
RCB-539 add missing file

Commit:
RCB-539: style cleanup
javadoc error ignore until delombok is in place

Commit:
RCB-539: more checkstyle

Commit:
RCB-539: more perm gen for java7

cp2boston commented on examples/src/main/java/com/basistech/rosette/examples/NameDeduplicationExample.java:
```diff
@@ -38,11 +39,11 @@ public static void main(String[] args) {
 
     private void run() throws IOException {
-        String nameDedupeData = "John Smith,Johnathon Smith,Fred Jones";
+        List<String> nameDedupeData = Arrays.asList("John Smith", "Johnathon Smith", "Fred Jones");
```
This will break the python code that populates the data string.  It looks for the variable name and replaces anything after the = with the new string.

cp2boston reviewed (changes requested):
I love the code reduction!  One small reversion.

Commit:
NameDeduplicationExample

- reverted to string representation of data to comply with slate update scripts

Commit:
RCB-539: use proper version

Commit:
Delombok

- to generate JavaDoc

javadoc modifications

- to point to delombok source

Commit:
RCB-539: purge unused classes

Commit:
Merge branch 'master' into rcb-539-lombok-mixin

Commit:
Fixed site:site compile

Commit:
Merged site.xml changes

Commit:
Potential winner!

- Generates model javadoc during mvn clean install
- Generates target/site/apidocs during mvn site:site

Commit:
lombok update to plugin

- Address PMD violations by not fully qualifying warning suppression

Commit:
RCB-539: lombok'ed batch models
since the backend uses it even though not public

Commit:
RCB-539: site/javadoc seems to work

Commit:
RCB-539: clean up a bit

Commit:
RCB-539: back to cp2's earlier stage

Commit:
RCB-539: urgly javadoc:aggreate workaround

Commit:
RCB-539: work around jackson builder problem

Commit:
RCB-539: same treatment for AdmReq

Commit:
RCB-539: bump version

Commit:
[maven-release-plugin] prepare release rosette-api-java-binding-1.8.104

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
Merge branch 'master' into rcb-539-lombok-mixin

Commit:
RCB-539: more updates
- changed from org.reflections to guava
- updated bnd to embedd bytebuddy, no shaing
- bring back a couple mixins

Commit:
RCB-539: use annotation to generate mixins

Commit:
RCB-539: fix pmd and checkstyle

Commit:
RCB-539: allow simple name input

Commit:
Merge branch 'master' into rcb-539-lombok-mixin
